sync.sh

```diff
@@ -0,0 +1,541 @@
+        printf "Error: %s is not installed, use -c/--command to specify.\n" "${COMMAND_NAME}" 1>&2
+        exit 1
+    fi
+
+    ARGS+=" -q "
+    SYNC_TIME_TO_SLEEP="3"
+    # Config file is created automatically after first run
+    if [[ -r ${CONFIG} ]]; then
+        # shellcheck source=/dev/null
+        source "${CONFIG}"
+        if [[ -n ${UPDATE_DEFAULT_CONFIG} ]]; then
+            printf "%s\n" "${CONFIG}" >| "${INFO_PATH}/google-drive-upload.configpath"
+        fi
+    fi
+    SYNC_TIME_TO_SLEEP="${TO_SLEEP:-${SYNC_TIME_TO_SLEEP}}"
+    ARGS+=" ${SYNC_DEFAULT_ARGS:-} "
+    "${UPDATE_DEFAULT_ARGS:-:}" DEFAULT_ARGS " ${ARGS} " "${CONFIG}"
+    "${UPDATE_DEFAULT_TIME_TO_SLEEP:-:}" SYNC_TIME_TO_SLEEP "${SYNC_TIME_TO_SLEEP}" "${CONFIG}"
+}
+
+###################################################
+# Triggers in case either -j & -k or -l flag ( both -k|-j if with positive integer as argument )
+# Priority: -j > -i > -l > -k
+# Globals: 5 variables, 6 functions
+#   Variables - JOB, SHOW_JOBS_VERBOSE, INFO_PID, LOG_PID, KILL_PID ( all array )
+#   Functions - _check_pid, _setup_loop_variables
+#               _kill_job, _show_jobs, _get_job_info, _remove_job
+# Arguments: None
+# Result: show either job info, individual info or kill job(s) according to set global variables.
+#   Script exits after -j and -k if kill all is triggered )
+###################################################
+_do_job() {
+    case "${JOB[*]}" in
+        *SHOW_JOBS*)
+            _show_jobs "${SHOW_JOBS_VERBOSE:-}"
+            exit
+            ;;
+        *PIDS*)
+            for pid in "${ALL_PIDS[@]}"; do
+                if [[ ${JOB_TYPE} =~ INFO ]]; then
+                    _get_job_info "${pid}" more
+                    status="${?}"
+                    if [[ ${status} != 0 ]]; then
+                        { [[ ${status} = 1 ]] && _remove_job "${pid}"; } || :
+                        printf "No job running with given PID ( %s ).\n" "${pid}" 1>&2
+                    fi
+                fi
+                if [[ ${JOB_TYPE} =~ SHOW_LOGS ]]; then
+                    input="$(grep "${pid}" "${SYNC_LIST}" || :)"
+                    if [[ -n ${input} ]]; then
+                        if _check_pid "${pid}"; then
+                            : "${input#*"|:_//_:|"}" && local_folder="${_/"|:_//_:|"*/}"
+                            _setup_loop_variables "${local_folder}" "${input/*"|:_//_:|"/}"
+                            tail -f "${LOGS}"
+                        fi
+                    else
+                        printf "No job running with given PID ( %s ).\n" "${pid}" 1>&2
+                    fi
+                fi
+                if [[ ${JOB_TYPE} =~ KILL ]]; then
+                    if [[ -n ${KILL_ALL} ]]; then
+                        PIDS="$(_show_jobs | grep -o 'PID:.*[0-9]' | sed "s/PID: //g" || :)" && total=0
+                        if [[ -n ${PIDS} ]]; then
+                            for pid in ${PIDS}; do
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This nested loop overrides the index variable of its parent.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5619177)